### DISCUSSION TITLE
Was Audio System Design Tool used in the code?

### DISCUSSION THREAD

**Discussion by radi8** [softerhardware/CWKeyer#45] [Q&A] [answered]
Was the audio system in the keyer designed with the Audio System Design Tool and was a mixer used in the design? I want to be able to shut off the audio from the receiver source while key down plus the hang time and figured that the best way might be to take the RX audio through an amplifier to a mixer where it can mix with the sine sidetone from the keyer and thence to the output (spkr, phones, line out). To mute the RX the amplifier gain can be set to zero (will there be clicks?). Where I have run into problems with the RX not being muted while keying is when using an online receiver like KiwiSDR and a second later I get my own morse back at me - very disconcerting. I know that there are other ways of muting the RX but the keyer knows about the keypresses and can work out hang times and has direct access to the audio. If the design tool was used, could I please have the connection diagram? This project is too good to be allowed to drift into history.

73, Graeme ZL2APV

**Comment by dl1ycf** [comment, marked as answer]
No graphical tool was used, the relevant commands have just been written down. The feature you are looking for is already there!
The CWKeyerShield class has a function set_cwptt_mute_option with which you can set this option, that is:

<cwkeyer>.cwptt_mute_option(1);

activates that feature. If activated, the RX audio (from USB) is not copied to the codec *while PTT is active*. 
This suppresses any unwanted side-tone from the radio.

Since you are using my TeensyWinkeyEmulator, just change  the config file config.keyershield.h line

#define MY_MUTE_OPTION                0   // set to 1 then RX audio is muted during CW PTT

and replace  "0" by 1. In the sketch, there is at some point the code

#ifdef MY_MUTE_OPTION
  cwshield.set_cwptt_mute_option(MY_MUTE_OPTION);
#endif


which does what you want. ATTENTION: even  if you are *not* using PTT outputs or MIDI signals, you *HAVE TO* enable
PTT in the winkey protocol, since the "mute on"/"mute off" conditions is set while switching PTT on and off.

Yours,

Christoph

**Reply by radi8** [reply]
Thank you Christoph, although you have already removed my need to ask the question hi.

73, Graeme
